cmake/std/atdm/contributed/kokkos-dev-2/environment.sh

```diff
@@ -0,0 +1,314 @@
+  if [[ -d /projects/sems/modulefiles ]] ; then
+    echo "NOTE: SEMS modules not defined, loading their definition!"
+    module use /projects/sems/modulefiles/projects
+    export SEMS_MODULEFILES_ROOT=/projects/sems/modulefiles
+  else
+    echo "ERROR: The SEMS modules are not defined and default location does not exist!"
+    return
+  fi
+fi
+
+module purge
+module load sems-env
+module use /home/projects/x86-64/modulefiles/local
+module load sems-git/2.10.1
+
+module load sems-cmake/3.12.2
+module load sems-ninja_fortran/1.8.2
+
+if [[ "$ATDM_CONFIG_NODE_TYPE" == "CUDA" ]] ; then
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=4
+  # We just need to be super conservative by default when using a GPU.  If
+  # users want to use more MPI processes, then can override this with
+  # ATDM_CONFIG_CTEST_PARALLEL_LEVEL_OVERRIDE.
+elif [[ "$ATDM_CONFIG_NODE_TYPE" == "OPENMP" ]] ; then
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=$(($ATDM_CONFIG_MAX_NUM_CORES_TO_USE/2))
+  export OMP_NUM_THREADS=2
+  # NOTE: With hyper-threading enabled, you can run as many threads as there
+  # are cores and with 2 OpenMP threads per MPI process, the means you can run
+  # as many MPI processes as there are cores on the machine with 2 OpenMP
+  # threads.  But we want to be conservative and instead run with half that
+  # many to be safe and avoid time-outs.
+else
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=$(($ATDM_CONFIG_MAX_NUM_CORES_TO_USE/2))
+  # NOTE: NOTE: When running in serial, the second hyperthread can't seem to
+  # run a sperate MPI process and if you try to run with as many they you get
+  # a bunch of failures that say "libgomp: Thread creation failed: Resource
+  # temporarily unavailable".  So we can only run with as many MPI processes
+  # as there are cores on the machine.  But we want to be conservative and
+  # instead run with half that many to be safe and avoid time-outs.
+  export OMP_PROC_BIND=FALSE
+  export OMP_NUM_THREADS=1
+fi
+
+if [[ "$ATDM_CONFIG_COMPILER" == "CLANG-7.0.1" ]] ; then
+  module load sems-clang/7.0.1
+  export OMPI_CXX=`which clang++`
+  export OMPI_CC=`which clang`
+  export OMPI_FC=`which gfortran`
+  export ATDM_CONFIG_LAPACK_LIBS="/usr/lib64/liblapack.so.3"
+  export ATDM_CONFIG_BLAS_LIBS="/usr/lib64/libblas.so.3"
+elif [[ "$ATDM_CONFIG_COMPILER" == "CLANG-9.0.0" ]] ; then
+  module load sems-clang/9.0.0
+  export OMPI_CXX=`which clang++`
+  export OMPI_CC=`which clang`
+  export OMPI_FC=`which gfortran`
+  export ATDM_CONFIG_LAPACK_LIBS="/usr/lib64/liblapack.so.3"
+  export ATDM_CONFIG_BLAS_LIBS="/usr/lib64/libblas.so.3"
+elif [[ "$ATDM_CONFIG_COMPILER" == "GNU-7.2.0" ]] ; then
+  module load sems-gcc/7.2.0
+  export OMPI_CXX=`which g++`
+  export OMPI_CC=`which gcc`
+  export OMPI_FC=`which gfortran`
+  export ATDM_CONFIG_LAPACK_LIBS="/usr/lib64/liblapack.so.3"
+  export ATDM_CONFIG_BLAS_LIBS="/usr/lib64/libblas.so.3"
+elif [[ "$ATDM_CONFIG_COMPILER" == "INTEL-17.0.1" ]] ; then
+  module load sems-intel/17.0.1
```

Comment:
@bartlettroscoe @e10harvey loading the sems-intel/17.* compilers may load an older gcc (or use system gcc), so it may be necessary in other atdm environment.sh scripts to swap the gcc module for a newer version that has full c++14 support to prevent issues as Trilinos moves to requiring c++14 (kokkos version 3.3 will require c++14)